app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ShoppingListController < ApplicationController
+  def index
+    @user = current_user
+    @foods = Food.all
+    @recipes = Recipe.where(user_id: @user.id)
+    @recipe_foods = RecipeFood.where(recipe_id: @recipes.ids)
+    @shopping_list = []
+    @recipe_foods.each do |recipe_food|
+      @shopping_list << recipe_food
+    end
+    @final_list = @foods.where.not(id: @shopping_list.map(&:food_id))
+
+    @price = @final_list.pluck(:price)
+    @quantity = @final_list.pluck(:quantity)
+    @total = @price.zip(@quantity).map { |x, y| x * y }.sum
```

Comment:
- [ ] when I click on generate shopping list it generates an empty list please make sure to fix that 
<img width="1345" alt="Screenshot 2022-12-16 at 11 02 19" src="https://user-images.githubusercontent.com/61790587/208075394-2016dac6-6f69-4763-bc6f-1d6230fc1daf.png">